.github/workflows/publish.yaml

```diff
@@ -60,6 +60,15 @@ jobs:
 
       - run: sudo apt install -y luarocks lua5.2 liblua5.2-dev libclang-common-14-dev clang-14
 
+      - run: sed -i -E
+          -e 's#branch = "master"#branch = "${{ github.head_ref || github.ref_name }}"#g'
```

Comment:
I'm not sure that this is what you want.
According to [the GitHub Actions documentation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs) about `github.ref_name`:
> For pull requests, the format is `<pr_number>/merge`.

Do you mean `github.event.pull_request.head.ref`?

Reply:
In PR `github.ref_name` is `ligurio/gh-27-fix-luarocks`, see https://github.com/ligurio/luzer/actions/runs/14336927442/job/40186276766?pr=33 and also https://gist.github.com/toshimaru/f14386ab69394e6ffb54aa799b9b124b